app/ide-desktop/lib/dashboard/src/layouts/TabBar.tsx

```diff
@@ -193,21 +196,13 @@ export function Tab(props: InternalTabProps) {
     }
   }, [observeElement])
 
-  React.useEffect(() => {
-    if (loadingPromise) {
-      setIsLoading(true)
-      loadingPromise.then(
-        () => {
-          setIsLoading(false)
-        },
-        () => {
-          setIsLoading(false)
-        }
-      )
-    } else {
-      setIsLoading(false)
-    }
-  }, [loadingPromise])
+  const { isLoading, data } = reactQuery.useQuery<backend.Project>(
```

Comment:
i really don't know whether it makes sense for a `Tab` of all things to overly specialize on `project`. is it perhaps possible to provide an optional `UseQueryOptions` instead?

Reply:
good idea, thanks for noticing!

Reply:
On the other hand, we can't decouple things here because right after we have:
```  typescript
const isFetching = isLoading || (data && data.state.type !== backend.ProjectState.opened)
```

Which couples things together. And I'm not sure how we can bypass this. maybe introduce another function that will only expose `isLoading`? or make a wrapper... Not sure 🤔

Ideally, if we want to make `tab` generic, we need to extract the `core` part into a separate component.

Reply:
i mean, we could have an optional override prop `isLoading: (query: T) => boolean`? which is... a bit ugly perhaps, it does *work*...

a more correct (?) way would probably be to make a *different* query with a `queryFn` that only resolves when the project is opened (and rejects if the project state becomes one of the closed states). not sure how feasible that is though?

Reply:
> different query with a queryFn that only resolves when the project is opened (and rejects if the project state becomes one of the closed states). not sure how feasible that is though?

Had a thought on this, but not sure how'd we cancel this query if we close the tab before the project is opened...